src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+        (*sleB)[sfMinAccountCreateAmount] = *minAccountCreate;
+    (*sleB)[sfXChainBridge] = bridge;
+    (*sleB)[sfXChainClaimID] = 0;
+    (*sleB)[sfXChainAccountCreateCount] = 0;
+    (*sleB)[sfXChainAccountClaimCount] = 0;
+
+    // Add to owner directory
+    {
+        auto const page = ctx_.view().dirInsert(
+            keylet::ownerDir(account), bridgeKeylet, describeOwnerDir(account));
+        if (!page)
+            return tecDIR_FULL;  // LCOV_EXCL_LINE [greg] test doesn't hit this?
+        (*sleB)[sfOwnerNode] = *page;
+    }
+
+    adjustOwnerCount(ctx_.view(), sleAcc, 1, ctx_.journal);
+
+    ctx_.view().insert(sleB);
+    ctx_.view().update(sleAcc);
+
+    return tesSUCCESS;
+}
+
+//------------------------------------------------------------------------------
+
+NotTEC
+BridgeModify::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;  // LCOV_EXCL_LINE
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const account = ctx.tx[sfAccount];
+    auto const reward = ctx.tx[~sfSignatureReward];
+    auto const minAccountCreate = ctx.tx[~sfMinAccountCreateAmount];
+    auto const bridge = ctx.tx[sfXChainBridge];
+
+    if (!reward && !minAccountCreate)
+    {
+        // Must change something
+        return temMALFORMED;
+    }
+
+    if (bridge.lockingChainDoor() != account &&
+        bridge.issuingChainDoor() != account)
+    {
+        return temSIDECHAIN_NONDOOR_OWNER;
+    }
+
+    if (reward && (!isXRP(*reward) || reward->signum() <= 0))
+    {
+        return temXCHAIN_BRIDGE_BAD_REWARD_AMOUNT;
+    }
+
+    if (minAccountCreate &&
+        ((!isXRP(*minAccountCreate) || minAccountCreate->signum() <= 0) ||
+         !isXRP(bridge.lockingChainIssue()) ||
+         !isXRP(bridge.issuingChainIssue())))
+    {
+        return temXCHAIN_BRIDGE_BAD_MIN_ACCOUNT_CREATE_AMOUNT;
+    }
+
+    return preflight2(ctx);
+}
+
+TER
+BridgeModify::preclaim(PreclaimContext const& ctx)
+{
+    auto const bridge = ctx.tx[sfXChainBridge];
+
+    if (!ctx.view.read(keylet::bridge(bridge)))
+    {
+        return tecNO_ENTRY;
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+BridgeModify::doApply()
+{
+    auto const account = ctx_.tx[sfAccount];
+    auto const bridge = ctx_.tx[sfXChainBridge];
+    auto const reward = ctx_.tx[~sfSignatureReward];
+    auto const minAccountCreate = ctx_.tx[~sfMinAccountCreateAmount];
+
+    auto const sleAcc = ctx_.view().peek(keylet::account(account));
+    if (!sleAcc)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    auto const sleB = ctx_.view().peek(keylet::bridge(bridge));
+    if (!sleB)
+        return tecINTERNAL;  // LCOV_EXCL_LINE
+
+    if (reward)
+        (*sleB)[sfSignatureReward] = *reward;
+    if (minAccountCreate)
+    {
+        // TODO: How do I modify minAccountCreate to clear it? With a flag?
```

Comment:
Fixed in f0f86aba7 [fold] Add a flag to clear the account create amount in the modify bridge txn

I went with a flag rather than treating "0" as special. We _may_ want to allow account create of zero at some point, and even if we don't if people try "0" it's going to do the opposite of the obvious behavior.